# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1203671136 18000
#      Fri Feb 22 04:05:36 2008 -0500
# Node ID 05eb3486f6507ef91ddc8bd0dd0722f6897208e9
# Parent  59bee17bc1923b7ff7b08c70b66da0ee85949ab0
__stepimp__: don't call subplot for single plot

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-02-22  John W. Eaton  <jwe@octave.org>
+
+	* control/base/__stepimp__.m: Don't use subplot for just one plot.
+
 2008-02-21  John W. Eaton  <jwe@octave.org>
 
 	* image/imshow.m: Call axis ("image").
 	From Michael G. Ross <mgross@MIT.EDU>.
 
 	* plot/errorbar.m: If nargout > 0, return vector of handles to
 	line objects.
 	* plot/__go_draw_axes__.m: Improve handling of line style for
diff --git a/scripts/control/base/__stepimp__.m b/scripts/control/base/__stepimp__.m
--- a/scripts/control/base/__stepimp__.m
+++ b/scripts/control/base/__stepimp__.m
@@ -232,17 +232,19 @@ function [y, t] = __stepimp__ (sitype, s
     else
       ssys = ss (F, G, C, D, t_step);
       gm = dcgain (ssys);
       tt = "step";
     endif
     ncols = floor (sqrt (NOUT));
     nrows = ceil (NOUT / ncols);
     for i = 1:NOUT
-      subplot (nrows, ncols, i);
+      if (nrows > 1 || ncols > 1)
+	subplot (nrows, ncols, i);
+      endif
       if (DIGITAL)
 	[ts, ys] = stairs (t, y(i,:));
 	ts = ts(1:2*n-2)';
 	ys = ys(1:2*n-2)';
 	if (length (gm) > 0)
 	  yy = [ys; gm(i)*ones(size(ts))];
 	else
 	  yy = ys;
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1268,17 +1268,17 @@ function __go_draw_axes__ (h, plot_strea
 
   else
     print_usage ();
   endif
 
 endfunction
 
 function [style, typ, with] = do_linestyle_command (obj, idx, mono,
-						    plot_stream, errbars)
+						    plot_stream, errbars = "")
 
   persistent have_newer_gnuplot ...
     = compare_versions (__gnuplot_version__ (), "4.0", ">");
 
   if (have_newer_gnuplot)
     fprintf (plot_stream, "set style line %d default;\n", idx);
   endif
   fprintf (plot_stream, "set style line %d", idx);
